atsign-foundation/at_server
# fix: version field in atMetadata not updating

**- What I did**
* Fix the issue of updating the key does not update the following fields in the metadata
   * version
   * createdBy
   * updatedBy

**- How I did it**
* Move the logic of incrementing the version to AtMetadataBuilder class.
* Pass atSign as an argument to AtMetadataBuilder class constructor to populate createdBy and updatedBy fields.
* Add a condition to prevent updating of "createdAt" field on update of a key. "createdAt" should be populated on the creation of a key and should remain unchanged. On updating the same key, "updatedAt" will updated.

**- How to verify it**
* Added the following unit tests to assert the changes:
  * A test to default field in metadata is set on a new key creation
  * A test to verify version field in metadata is set to 1 on updating the existing key
  * A test to verify version field in metadata is set to 1 when updating metadata using putMeta method
  * A test to verify version field in metadata is set to 1 when using putAll method
  
* Update the sync unit tests and functional tests to assert the changes. The changes are in the following PR: https://github.com/atsign-foundation/at_server/pull/1158

**- Description for the changelog**
* AtMetadata.version does not update on the update of a key
